test/server/http/admin_test.cc

```diff
@@ -206,30 +216,68 @@ TEST_P(AdminInstanceTest, CustomHandler) {
   EXPECT_EQ(Http::Code::Accepted, getCallback("/foo/bar", header_map, response));
 }
 
-TEST_P(AdminInstanceTest, RejectHandlerWithXss) {
-  auto callback = [](absl::string_view, Http::HeaderMap&, Buffer::Instance&) -> Http::Code {
-    return Http::Code::Accepted;
+TEST_P(AdminInstanceTest, AdminStream) {
+  // Cover endStreamOnComplete().
+  // Add an handler that sets end_stream_on_complete on the AdminStream object.
+  auto callback = [](absl::string_view, Http::HeaderMap&, Buffer::Instance&,
+                     AdminStream& admin_stream) -> Http::Code {
+    admin_stream.setEndStreamOnComplete(false);
+
+    // check that value is set to false
+    if (admin_stream.endStreamOnComplete())
+      return Http::Code::NotAcceptable;
+    return Http::Code::OK;
   };
+
+  // Add non removable handler.
+  EXPECT_TRUE(admin_.addHandler("/foo/bar", "hello", callback, false, false));
+  Http::HeaderMapImpl header_map;
+  Buffer::OwnedImpl response;
+  EXPECT_EQ(Http::Code::OK, getCallback("/foo/bar", header_map, response));
+
+  // Cover getDecoderFilterCallbacks() method.
+  auto method = Http::Headers::get().MethodValues.Get;
+  request_headers_.insertMethod().value(method.data(), method.size());
+  NiceMock<Http::MockStreamDecoderFilterCallbacks> callbacks;
+  ON_CALL(callbacks, streamId()).WillByDefault(testing::Return(1234));
+  std::list<std::function<void()>> on_destroy_callbacks{};
+  AdminStreamImpl admin_stream(callbacks, request_headers_, on_destroy_callbacks);
+
+  // Instead of the double cast, maybe streamId() can be changed to const method?
+  Http::MockStreamDecoderFilterCallbacks& got_callbacks =
+      const_cast<Http::MockStreamDecoderFilterCallbacks&>(
+          dynamic_cast<const Http::MockStreamDecoderFilterCallbacks&>(
+              admin_stream.getDecoderFilterCallbacks()));
+  EXPECT_EQ(got_callbacks.streamId(), 1234);
+
+  // Cover getRequestHeaders() method.
+  const Http::TestHeaderMapImpl temp_request_headers;
+
+  const Http::TestHeaderMapImpl got_request_headers = admin_stream.getRequestHeaders();
+  EXPECT_EQ(got_request_headers.byteSize(), request_headers_.byteSize());
+}
+
+TEST_P(AdminInstanceTest, RejectHandlerWithXss) {
+  auto callback = [](absl::string_view, Http::HeaderMap&, Buffer::Instance&,
```

Comment:
You could factor this helper callback generator into a simple static test method, since it is used 3x (previously 2x)